src/apis/btc-core.ts

```diff
@@ -1,31 +1,56 @@
-import { BlockApi, Status, TxApi } from "@interlay/esplora-btc-api";
+import { BlockApi, Status, TxApi, ScripthashApi, Transaction } from "@interlay/esplora-btc-api";
 import { AxiosResponse } from "axios";
+import * as bitcoinjs from "bitcoinjs-lib";
 
 const mainnetApiBasePath = "https://blockstream.info/api";
-const testnetApiBasePath = "https://blockstream.info/testnet/api";
+const testnetApiBasePath = "https://electr-testnet.do.polkabtc.io";
 
 export type TxStatus = {
     confirmed: boolean,
     confirmations: number,
 }
 
+export type TxOutput = {
+    scriptpubkey: string,
+    scriptpubkeyAsm: string,
+    scriptpubkeyType: string,
+    scriptpubkeyAddress: string,
+    value: number
+}
+
+export type TxInput = {
+    txId: string,
+    vout: number,
+    isCoinbase: boolean,
+    scriptsig: string,
+    scriptsigAsm: string,
+    innerRedeemscriptAsm: string,
+    innerWitnessscriptAsm: string,
+    sequence: number,
+    witness: string[],
+    prevout: TxOutput
+}
+
 export interface BTCCoreAPI {
     getLatestBlock(): Promise<string>;
     getLatestBlockHeight(): Promise<number>;
     getMerkleProof(txid: string): Promise<string>;
     getTransactionStatus(txid: string): Promise<TxStatus>;
     getTransactionBlockHeight(txid: string): Promise<number | undefined>;
     getRawTransaction(txid: string): Promise<Buffer>;
+    getTxByOpcode(opcode: string): Promise<string>;
```

Comment:
can you rename to `getTxIdByOpcode` since we don't want to return the full transaction but rather just the id.